permission-portal-cloud-functions/functions/src/index.ts

```diff
@@ -267,6 +292,37 @@ export const createUser = functions.https.onCall((newUser, context) => {
   });
 });
 
+// Callable for admins to delete users in their organization.
+// Recall that firestore rules are made irrelevant by our use of the admin client, so we need to check ourselves
+// that the caller is an admin for the to-be-deleted user's organization (using isUserInCallersOrganizationGuard())
+export const deleteUser = functions.https.onCall((data, context) => {
+  return new Promise((resolve, reject) => {
+    isUserInCallersOrganizationGuard(data.email, context)
+      .then(() => {
+        auth
+          .getUserByEmail(data.email)
+          .then((userRecord) => {
+            deleteUserByUid(userRecord.uid)
+              .then(() => {
+                resolve(`Successfully deleted user ${data.email}`);
+              })
+              .catch((err) => {
```

Comment:
Afaik the exceptions technically do bubble out (if you don't handle them they get thrown). The default `tslint` rule that's enforcing this endless chain of catches is
```json
// Functions must either be handled directly (e.g. with a catch() handler) or returned to another function.
    // This is a major source of errors in Cloud Functions and the team strongly recommends leaving this rule on.
    "no-floating-promises": true,
```

Because the comment seems to so heavily recommend it, I've just been going with it.